ui/test/unit/reports/controllers/reportsController.spec.js

```diff
@@ -94,18 +102,42 @@ describe("ReportsController", function () {
 
     it('should initialise date range with supportedDateRange config', function () {
         rootScope.default.reportsRequiringDateRange = {
-            dateRangeType: new Date(),
-            startDate: new Date(),
-            stopDate: new Date(),
+            dateRangeType: new Date()
         };
         scope.setDefault('dateRangeType', 'reportsRequiringDateRange');
-        scope.setDefault('startDate', 'reportsRequiringDateRange');
-        scope.setDefault('stopDate', 'reportsRequiringDateRange');
 
         expect(_.keys(rootScope.default.reportsRequiringDateRange).length).toBe(3);
         expect((rootScope.default.reportsRequiringDateRange.dateRangeType).getDate()).toBe(new Date().getDate());
-        expect(rootScope.reportsRequiringDateRange[0].startDate).toBe(rootScope.default.reportsRequiringDateRange.startDate);
-        expect(rootScope.reportsRequiringDateRange[0].stopDate).toBe(rootScope.default.reportsRequiringDateRange.stopDate);
+        expect(rootScope.reportsRequiringDateRange[0].startDate.getDate()).toBe(new Date().getDate());
+        expect(rootScope.reportsRequiringDateRange[0].stopDate.getDate()).toBe(new Date().getDate());
+    });
+
+    const previousMonthTestCases = [
```

Comment:
Since all these scenarios have been added into a single datastructure, when this test fails, how does one know which use-case failed? Is there some log which prints the expected/calculated date?

Reply:
@gsluthra , when test case fails it would print the specific test case name for example- " should return previous date and month when ${currentDate}`" . Here current date would be the specific date.

Reply:
No, maybe this is sufficient.